matrix/matrix.py

```diff
@@ -0,0 +1,202 @@
+import operator
+from collections.abc import Iterable
+
+
+class Matrix:
+    """An immutable object representing a matrix data-type.
+
+    Attributes:
+        _matrix (tuple): contains the data about the matrix. Each line of
+            the matrix is represented by a tuple.
+    Note:
+        This implementation only supports even-side matrices (n * n).
+    """
+
+    def __init__(self, *lines):
+        self._validate(*lines)
+        self._matrix = tuple(lines)
+
+    def _validate(self, *data: tuple):
+        """Check if the data given is valid.
+
+        Note:
+            Valid data is defined if it contains tuples of the same length
+            (Matrix only supports n * n), items can't be iterable.
+        """
+        if not all(isinstance(line_data, tuple) for line_data in data):
+            raise ValueError("Matrix lines must be tuple")
+
+        if any(isinstance(item, Iterable) for line_data in data for item in
+               line_data):
+            raise ValueError("Matrix lines must be tuple")
+
+        if not all([len(line) == len(data) for line in data]):
+            raise ValueError("Matrix must be even-sided")
+
+    @classmethod
+    def unity(cls, dimension: int):
+        """Generates a matrix with a main diagonal of 1's, everything else
+        is 0's.
+        """
+        return cls(*[tuple((0,) * j + (1,) + (0,) * i) for j, i in enumerate(
+            range(dimension - 1, -1, -1))])
+
+    @classmethod
+    def ones(cls, dimension):
+        """Generates a matrix full of 1's."""
+        return cls(*([tuple((1,) * dimension) for _ in range(dimension)]))
+
+    def __repr__(self):
+        return super(Matrix, self).__repr__()
```

Comment:
the purpose here, is not that you will use already existing repr, but that you will create one yourself